### ISSUE TITLE
Working Directory for new sessions not go to previous session's directory

### ISSUE THREAD

**Issue by kankunnawat** [warpdotdev/Warp#3045]
### Discord username (optional)

kan#3740

### Describe the bug

Working Directory for new sessions not go to previous session's directory even though i choose that option, instead it always goes to home directory.
I tried toggle it with home directory and go back to previous session's directory but it still doesn't work.

### To Reproduce

1. Open warp
2. go to some directory
3. Close warp
4. It always goes to home directory

### Expected behavior

It should go to previous session's directory

### Operating System

MacOS

### Operating System and Version

13.3.1 (a) 

### Shell Version

zsh 5.9 (x86_64-apple-darwin22.0)

### Warp Version

v0.2023.05.02.08.03.stable_00

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Hi @kankunnawat sorry about this blocker, may you please review the following doc on setting Working Directory and let us know if it helps resolve your issue? https://docs.warp.dev/features/working-directory

**Comment by kankunnawat** [comment]
> Hi @kankunnawat sorry about this blocker, may you please try running the following and let us know if it helps resolve your issue, run `defaults delete dev.warp.Warp-Stable WorkingDirectoryConfig` then Quit Warp with `CMD-Q`, then reopen and reconfigure your Working Directory.
> 
> https://docs.warp.dev/features/working-directory

I've tried it but still not solve the issue.

**Comment by kankunnawat** [comment]
I noticed some patterns going on after i run your command.
If I _quit_ the warp by `CMD+Q` It'll return back to previous session's directory when openning again, but if i just _close_ it by `CMD+W` It'll always go to home directory. 
Before both way always work as expected which is going to whatever option you choose(in my case it's previous session's directory.

**Comment by dannyneira** [comment]
Thanks for letting me know, that makes sense now, you have to disable Session Restoration for that behaviour to stop. When you fully close Warp with CMD-Q, it will restore all previous windows/tabs/panes and their respective CWDs, vs closing with CMD-W just closes the window, not fully quit Warp, and so opening a new window will open in the Working directory you setup. Please see steps here, and let me know if that helps. https://docs.warp.dev/features/sessions/session-restoration

**Comment by kankunnawat** [comment]
After i turn off that feature, now it doesn't come back to previous's session directory for both `cmd+Q or cmd+W`

<img width="1136" alt="image" src="https://github.com/warpdotdev/Warp/assets/40234606/203ca99e-6847-4d35-86a4-626d186f5581">

**Comment by dannyneira** [comment]
@kankunnawat May you please confirm the above, that you have the setting set to previous sessions directory, and also include a video of the issue so that we can try and narrow down the root cause?

**Comment by dannyneira** [comment]
Closing as there is no response/confirmation of the above, but please reopen if you can confirm and reproduce the issue with the [above settings](https://github.com/warpdotdev/Warp/issues/3045#issuecomment-1544248521).

**Comment by kankunnawat** [comment]
Yes, it's already and always previous sessions directory. 
<img width="683" alt="image" src="https://github.com/warpdotdev/Warp/assets/40234606/dc954ed5-a036-4076-9402-e9869fb6b4a2">
Regarding to how to produce the issue, i already explain in [above comment](https://github.com/warpdotdev/Warp/issues/3045#issuecomment-1541304706). You can also find the video of how to reproduce [here](https://drive.google.com/file/d/1hJmg6m1PJfbWA1-0zbvMFbuMzfMEe69K/view?usp=sharing).

**Comment by kankunnawat** [comment]
that's not true. This feature "Restore windows, tabs, and panes on startup" already enable 
<img width="685" alt="image" src="https://github.com/warpdotdev/Warp/assets/40234606/214c6615-e288-4ec2-ae86-291e85f533d1">

**Comment by vorporeal** [comment]
Interesting, then I expect there's something wrong with the database file where we store the session restoration data.

Can you run `rm /Users/$USER/Library/Application\ Support/dev.warp.Warp-Stable/warp.sqlite`, close and re-open Warp, then test this out again?

**Comment by kankunnawat** [comment]
I tried running that command, still get the same behavior as in this [comment](https://github.com/warpdotdev/Warp/issues/3045#issuecomment-1541304706)

**Comment by kankunnawat** [comment]
so any update?

**Comment by dannyneira** [comment]
You could try resetting settings in Warp with `defaults delete dev.warp.Warp-Stable`

Learn more [here](https://docs.warp.dev/help/uninstalling-warp#removing-warp-login-settings-files-log-and-database)

**Comment by kankunnawat** [comment]
Same

**Comment by dannyneira** [comment]
Heard, we've reopened the issue and will look into this, Once we have updates, we'll reach out on this thread. 

Thanks for your patience while we try and work out this issue.

**Comment by louiss0** [comment]
I'm having this problem too! Please prioritize this type of issue! PLEASE! 
This is so bad! Now I'll have to get into the habit of closing all sessions before I close the app!

**Comment by yogeek** [comment]
I confirm, I was about to open an issue for this but I found this one.
Loosing currend directory is really a big issue when working with AI agents like opencode and claude code which stores sessions in different folders. Using agents, restoring windows, tabs and paned are not sufficient, CWD is mandatory to get a frictionless experience.

**Comment by louiss0** [comment]
> I confirm, I was about to open an issue for this but I found this one.
> Loosing currend directory is really a big issue when working with AI agents like opencode and claude code which stores sessions in different folders. Using agents, restoring windows, tabs and paned are not sufficient, CWD is mandatory to get a frictionless experience.

Please send logs Please 🙏. I don't know how to experiment!

**Comment by louiss0** [comment]
Here's the log I noticed something strange!  [warp.log](https://github.com/user-attachments/files/26910079/warp.log)

```log
2026-04-20T20:13:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:13:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:13:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:13:01Z [INFO] active window changed: None
2026-04-20T20:13:01Z [INFO] dispatching global action for root_view:update_quake_mode_state
2026-04-20T20:13:01Z [INFO] dispatching global action for workspace:save_app
2026-04-20T20:13:24Z [INFO] Start to send telemetry events to RudderStack
2026-04-20T20:14:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:14:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:14:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:15:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:15:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:15:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:16:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:16:00Z [WARN] Failed to convert path: Given path was not a UNIX path
2026-04-20T20:16:00Z [WARN] Failed to convert path: Given path was not a UNIX path
```

**Comment by NoNameProvided** [comment]
This seems to be fixed for me now in the latest release. 

The only relevant fix I can see was added in https://github.com/warpdotdev/warp/pull/9277, which fixes a bug where `~` was not expanded correctly for new tabs. Our problem reported here is not for new tabs but the code may be reused between new tabs and reopening tabs.

**Comment by thomast8** [comment]
Still happening on Warp 0.2026.05.20. "Previous session's directory" works for a plain new tab, but a tab opened from a tab/launch config always opens in $HOME and ignores it, so config-launched tabs can't inherit the repo you were in. Would love for configs to honor the setting too.
